extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/RestClientBuilderHelper.java

```diff
@@ -112,4 +110,58 @@ public static Sniffer createSniffer(RestClient client, ElasticsearchConfig confi
 
         return builder.build();
     }
+
+    public enum EsAuth {
+        NONE {
+            @Override
+            public void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config) {
+                // No authentication needed
+            }
+        },
+
+        BASIC {
+            @Override
+            public void apply(HttpAsyncClientBuilder httpClientBuilder, ElasticsearchConfig config) {
+                if (config.username().isPresent()) {
```

Comment:
since you are returning an auth type after you've checked the presence of the properties these very outer `if`s aren't necessary anymore, right?